src/container/srv_container.c

```diff
@@ -4665,48 +4687,24 @@ ds_cont_set_label(struct cont_svc *svc, uuid_t uuid, daos_prop_t *prop_in, bool
 
 		/* Update prop IV with merged prop */
 		rc = cont_iv_prop_update(pool->sp_iv_ns, uuid, prop_iv);
-		if (rc != 0) {
-			D_ERROR(DF_UUIDF" failed to update prop IV for cont: "DF_RC"\n",
-				DP_UUID(uuid), DP_RC(rc));
-			goto out_cont;
-		}
 	} else {
-		d_iov_set(&key, prop_in->dpp_entries[0].dpe_str,
-			  strnlen(prop_in->dpp_entries[0].dpe_str, DAOS_PROP_MAX_LABEL_BUF_LEN));
-		d_iov_set(&val, tmp, sizeof(uuid_t));
-		rc = rdb_tx_lookup(&tx, &cont->c_svc->cs_uuids, &key, &val);
-		if (rc == 0) {
-			/* If specified label matches existing label in svc, do nothing. */
-			if (uuid_compare(uuid, tmp) == 0)
-				D_GOTO(out_cont, rc = 0);
-
-			D_ERROR("Forbid non-unique label (%s) for different containers: "
-				DF_UUIDF"/"DF_UUIDF"\n", prop_in->dpp_entries[0].dpe_str,
-				DP_UUID(uuid), DP_UUID(tmp));
-			D_GOTO(out_cont, rc = -DER_EXIST);
-		} else if (rc == -DER_NONEXIST) {
-			d_iov_set(&val, uuid, sizeof(uuid_t));
-			rc = rdb_tx_update(&tx, &cont->c_svc->cs_uuids, &key, &val);
-			if (rc != 0) {
-				D_ERROR(DF_UUIDF" failed to set container label %s: "DF_RC"\n",
-					DP_UUID(uuid), prop_in->dpp_entries[0].dpe_str, DP_RC(rc));
-				goto out_cont;
-			}
-		} else {
-			D_ERROR(DF_UUIDF" failed to lookup container label %s in svc: "DF_RC"\n",
-				DP_UUID(uuid), prop_in->dpp_entries[0].dpe_str, DP_RC(rc));
-			goto out_cont;
-		}
+		rc = check_set_prop_label(&tx, cont, prop_in, prop_old);
 	}
 
 out_cont:
+	if (rc == 0)
+		rc = rdb_tx_commit(&tx);
 	cont_put(cont);
```

Comment:
[Not requesting changes] Normally we would write the `rdb_tx_commit` call above `out_cont`, so that the `rc = 1` trick would not be required, no?

Reply:
OK